full-check.sh

```diff
@@ -12,8 +12,11 @@ printf "========================= MYPY STATIC TYPE CHECKER =================\n"
 mypy --config-file=mypy.ini concent_api/
 printf "\n"
 
-printf "============================ UNIT TESTS ============================\n"
+printf "========================= UNIT TESTS WITH COVERAGE =================\n"
 # NOTE: 'manage.py test' does not find all tests unless we run it from within the app directory.
+rm -rf contrib/coverage/coverage_html
 cd concent_api/
-python3 manage.py test
+coverage run --rcfile=../coverage-config --source='.' manage.py test
+coverage html --rcfile=../coverage-config
```

Comment:
I'd remove the html report from this script. There's no point in generating it every time we run checks of the code. The report in the console is enough. I think that most of the time the programmer will not even look at the html report.

It may be useful to generate it from time to time and look at it but I think there should be a separate script for that. Or just run the command manually.

Reply:
:+1:

Reply:
Depends on what we are trying to achieve, if we want to push only 100% tested code then it might be useful, if not then it is ok to remove imho.

Reply:
But the summary in the console already shows coverage? HTML report is just a nicely formatted version of it, right? So it's not necessary to generate it every time. Only if you want to look at it.

Reply:
No, HTML report shows a list of html files, where each other is a separate file that was examined for coverage, and it marks not covered lines in red, covered lines in green. It also show some extra information, like for example if you have
`
if something:
    do()
`
if both paths (True and False) were taken. Go ahead and generate one and take a look. It is very useful if you want to check which lines exactly are not covered. So, please let me know if you want to keep it or not. It could take a while to generate if the project gets really big, but how often does developer runs full_check ? If he is stuck on test or lint, he can run only a single command.

So please

Reply:
OK. I haven't looked at the details of the report. I like it.

I just don't think `full_check.sh` is a good place for it. It's meant to be comprehensive but still pretty cheap to run during development. If use it when I want to run all the tests because extra checks do not take all that long and are useful. Only when I start debugging a specific problem I start running `lint.sh` or specific unit test. I don't think we'll be looking at the report just as often.

That being said, I think looking at this detailed report is pretty useful. I suggest making a separate script for generating it. I was also thinking about serving the coverage report from the development cluster though I'm not sure how useful it would be. Maybe not that useful because we should check coverage before we even let the code run on the cluster. Anyway, if you have any other idea on when to run it, we can discuss it.